utils/land-patterns.stanza

```diff
@@ -18,6 +20,55 @@ defn ensure-even-positive! (value: Int, name: String):
   if value < 2 or value % 2 == 1:
     fatal("%_ only supports a positive even number of pins." % [name])
 
+; ====== Silk screen helpers ===================================================
+; Create a polarity marker relative to a pad.
+; pda
+public defn draw-polarity-marker (pad-1:JITXObject, part-dims?:Dims|False) :
+  val min-silk-width = clearance(current-rules(), MinSilkscreenWidth)
+  inside pcb-landpattern:
+    ; draw a line of `min-silk-width` width and length 3.0 * min-silk-width 
+    ; centered at origin
+    defn draw-line (origin:Pose):
+      val line-width  = min-silk-width
+      val line-length = line-width * 3.0
+      val line = Line(line-width, [
+        Point(min-silk-width * 0.5, 0.0)
+        Point(line-width + min-silk-width * 0.5, 0.0)  
+      ])
+      layer(Silkscreen("pol", Top)) = origin * line
+
+    defn part-pose () :
+      match(part-dims?) :
+        (_:False): 
+          loc(0.0, 0.0)
+        (d:Dims): 
+          loc(x(d) * -0.5,
+              y(d) *  0.5 + min-silk-width * 0.5 + PLACEMENT-TOLERANCE)
+
+    defn distance (p:Pose):
+      val [x, y] = [x(center(p)), y(center(p))]
+      sqrt(x * x + y * y)
+    
+    defn pad-pose ():
+      val pad-loc    = pose(pad-1)
+      val pad-dims   = dims(pad-shape(def-pad(pad-1)))
+      val R = SOLDER-MASK-REGISTRATION + min-silk-width * 0.5 + SILK-SOLDER-MASK-CLEARANCE + 0.01
+      loc(
+        x(pad-dims) * -0.5 + x(center(pad-loc)) - SOLDER-MASK-REGISTRATION
+        y(pad-dims) *  0.5 + y(center(pad-loc)) + R
+      )
+    
+    val [p1, p2] = [pad-pose(), part-pose()] 
+    draw-line $
+      if distance(p1) > distance(p2):
+        p1
+      else: 
+        p2
```

Comment:
max(distance, [p1 p2])?

Reply:
Which usage of `max` is this? I'm not familiar

Reply:
or maybe `maximum`